src/main/java/com/github/wvengen/maven/proguard/ProGuardMojo.java

```diff
@@ -48,132 +48,118 @@
 import org.apache.maven.plugin.MojoExecutionException;
 import org.apache.maven.plugin.MojoFailureException;
 import org.apache.maven.plugin.logging.Log;
+import org.apache.maven.plugins.annotations.Component;
```

Comment:
Note that the `Component` annotation is marked as deprecated. It seems the standard `@Inject` annotations should be used instead? See https://maven.apache.org/maven-jsr330.html#how-to-use-jsr-330-in-plugins

But I am not familiar with how this works with Maven (and whether the `javax` or Jakarta annotations can / should be used). Maybe for now this can keep using this deprecated annotation. Might already be an improvement over the previous Javadoc based variant.